packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -170,18 +173,21 @@ class _AddBasicDetailsPageState extends State<AddBasicDetailsPage> {
                     //And if it is not, we create a new product with the fields of the _product
                     // and we insert it in the database. (Giving the user an immediate feedback)
                     if (product == null) {
-                      daoProduct.put(Product(
+                      final Product newProduct = Product(
```

Comment:
This I don't like: please create a method that populates the 3 fields, and call it twice. Same behavior as for the other pages.